modules/test/playwright/tests/object-web/objectDefinitions.spec.ts

```diff
@@ -440,3 +457,62 @@ test.describe('Manage object definitions through View Object Definitions', () =>
 		);
 	});
 });
+
+test.describe('Manage object definitions through a Page', () => {
+	test('can display an object reactivated on the Page Item Selector', async ({
+		apiHelpers,
+		page,
+		pageEditorPage,
+		site,
+		viewObjectDefinitionsPage,
+	}) => {
+		const objectDefinition =
+			await apiHelpers.objectAdmin.postRandomObjectDefinition({
+				objectFolderExternalReferenceCode: 'default',
+				status: {code: 0},
+			});
+
+		await viewObjectDefinitionsPage.goto();
+
+		await viewObjectDefinitionsPage.changeObjectActivateStatus(
+			objectDefinition.name
+		);
+
+		await viewObjectDefinitionsPage.goto();
+
+		await viewObjectDefinitionsPage.changeObjectActivateStatus(
+			objectDefinition.name
+		);
+
+		const headingDefinition = getFragmentDefinition({
+			id: getRandomString(),
+			key: 'BASIC_COMPONENT-heading',
+		});
+
+		const layout = await apiHelpers.headlessDelivery.createSitePage({
+			pageDefinition: getPageDefinition([headingDefinition]),
+			siteId: site.id,
+			title: getRandomString(),
+		});
+
+		await pageEditorPage.goto(layout, site.friendlyUrlPath);
+
+		await pageEditorPage.selectFragment(headingDefinition.id);
+
+		await page.getByLabel('Select element-text').click();
+
+		await page.getByLabel('Select Item').click();
+
+		await expect(
+			page
+				.frameLocator('iframe[title="Select"]')
+				.getByRole('menuitem', {name: objectDefinition.name})
+		).toBeVisible();
+
+		// Clean up
+
+		await apiHelpers.objectAdmin.deleteObjectDefinition(
+			objectDefinition.id
+		);
```

Comment:
Hey @vngoliveira, could you improve the clean up of this entire file? We are trying to remove this type of cleanup because it is not safe for this cleanup to be performed, if for some reason this test fails in the middle, this cleanup will not be performed.

So I suggest you do something similar to what we did in objectField.spec.ts. In this Acase from what I can see, just create a let containing all the objectDefinitions created in each test at the beginning of the file, and then create a test.afterEach to delete the definitions:

```ts
let objectDefinitions: ObjectDefinition[] = [];

test.afterEach(async ({apiHelpers}) => {
	if (objectDefinitions.length) {
		for (const objectDefinition of objectDefinitions) {
			await apiHelpers.listTypeAdmin.deleteListTypeDefinition(
				objectDefinition.id
			);
		}

		objectDefinitions = [];
	}
});
```

And in each test you will fill objectDefinitions with the created object definition(s). You can create a new commit to do this.